src/syscheckd/syscheck_audit.c

```diff
@@ -671,19 +675,24 @@ void audit_parse(char *buffer) {
                     w_evt->audit_name = NULL;
                     w_evt->audit_uid = NULL;
                 } else {
-                    const char *user = get_user("",atoi(auid), NULL);
+                    char *user = get_user("",atoi(auid), NULL);
                     w_evt->audit_name = strdup(user);
                     w_evt->audit_uid = strdup(auid);
+                    os_free(user);
                 }
                 os_free(auid);
             }
             // effective_name && effective_uid
             if(regexec(&regexCompiled_euid, buffer, 2, match, 0) == 0) {
                 match_size = match[1].rm_eo - match[1].rm_so;
-                os_malloc(match_size + 1, w_evt->effective_uid);
-                snprintf (w_evt->effective_uid, match_size + 1, "%.*s", match_size, buffer + match[1].rm_so);
-                const char *user = get_user("",atoi(w_evt->effective_uid), NULL);
-                w_evt->effective_name = strdup(user);
+                char *euid = NULL;
+                os_malloc(match_size + 1, euid);
+                snprintf (euid, match_size +1, "%.*s", match_size, buffer + match[1].rm_so);
+                char *user = get_user("",atoi(euid), NULL);
+                os_strdup(user, w_evt->effective_name);
```

Comment:
We don't need to allocate memory new memory. Please, assign the output of `get_user` to ` w_evt->effective_name)` and remove the user variable.